hardware/OpenZWave.cpp

```diff
@@ -1344,8 +1344,18 @@ bool COpenZWave::SwitchLight(_tZWaveDevice* pDevice, const int instanceID, const
 			if (GetValueByCommandClassIndex(pDevice->nodeID, (uint8_t)instanceID, COMMAND_CLASS_SWITCH_MULTILEVEL, ValueID_Index_SwitchMultiLevel::Level, vID) == true)
 			{
 				std::string vLabel = m_pManager->GetValueLabel(vID);
+                
+                // Correct "open" value for Fibaro FGR-223 (Roller Shutter 3)
+                if ( (pDevice->Manufacturer_id == 0x010F) && (pDevice->Product_id == 0x1000) && (pDevice->Product_type == 0x0303) ) {
+                    if ( svalue == 255 ) {
```

Comment:
255 is something else then 100 ?
I have not received any problems concerning operation of roller shutters and did verify the operation a (long) time ago

Reply:
Last time I was checking 255 was set for Open for dimmable/percentage devices. Actually don't know why not 100(maybe because percentage bind is being treated the same way as dimmable light ?), but a few lines above just suggest it is being used that way. Anyway setting 100 is wrong since in ZWave docs this value is for "return to a previous position". You can find posts on the forum reporting that in a situation when we first move a blind to some position (i.e 50%) then close it and than try to open it (domoticz sets 100% for open right now), than the blind goes back to 50% instead Open position.
I'm not familiar with the whole domoticz code, so probably there is a better way to fix it :) but this one works as expected too :)

Reply:
In mainworker.cpp there is also the following code snippet:

```
		else if (((switchtype == STYPE_BlindsPercentage) ||
			(switchtype == STYPE_BlindsPercentageInverted)) &&
			(gswitch.cmnd == gswitch_sSetLevel) && (level == 100))
			gswitch.cmnd = gswitch_sOn;

```

You have to send the switch type to 'blinds', but did you test/debug the value in the console/log when you set the slider all the way to the left/right ?